bson/src/main/org/bson/codecs/pojo/annotations/BsonRepresentation.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.pojo.annotations;
+
+import org.bson.BsonType;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that specifies what type the property is stored as in the database.
+ *
+ * <p>Note: Requires the {@link org.bson.codecs.pojo.Conventions#ANNOTATION_CONVENTION}</p>
+ *
+ * @since 4.2
+ * @see org.bson.codecs.pojo.Conventions#ANNOTATION_CONVENTION
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
```

Comment:
Should we be targeting anything besides FIELD here?  I notice that some of the other annotations target METHOD, for example.

Reply:
We could definitely target METHOD. Imo that is the only other `ElementType` that would make any sense to target. I chose not to because of this in our docs:

"Annotations can be applied to read and / or write contexts by configuring the getter / setter methods. Any annotations applied to a field are applied to both read and write contexts."

`BsonRepresentation` applies to both reading and writing, so I didn't want to let users only annotate a getter or setter and have the impression that it only affected reading/writing. Also to me it just seems cleaner to annotate the field as opposed to the getter/setter if you're not trying to distinguish between reading and writing. Do you agree with that rationale, or do you still think we should allow it to target methods as well?

Reply:
@rozza, interested in your opinion here.  I'm not sure why BsonProperty, for example, applies to field, method, and parameter, and what happens if there is disagreement between them.  Maybe it's an error to add it to more than one place.

Reply:
If field naming doesn't match the convention then setting on getters and setters would be the only way for a user to apply the annotation equally.  So I would add to METHOD as well.

BsonProperty's are special and can also be applied to parameters - when used in constructors, it allows the mapping of arguments to properties.  There can be conflicting read and write properties as models don't necessarily have to be symmetrical (See the `AsymmetricalModel` examples).

Reply:
Okay sounds good I'll add it.